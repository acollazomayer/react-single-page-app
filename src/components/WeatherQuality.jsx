import React, { Component } from 'react';
import { Panel } from 'react-bootstrap';
import PropTypes from 'prop-types';
import PropertyRow from './PropertyRow';
import Spinner from './Spinner';
import en from '../locales/en';
import '../styles/airquality.css';


class WeatherQuality extends Component {
  renderComponentBody() {
    if (this.props.error) return this.props.error.message;
    if (this.props.isFetching) {
      return (
        <Spinner />
      );
    }
    const {
      breezometerAqi,
      breezometerDescription,
      dominantPollutantCanonicalName,
      dominantPollutantDescription,
    } = { ...this.props.weatherState };

    return (
      <div>
        <PropertyRow
          title={en.AirQualityMessage}
          message={breezometerDescription}
        />
        <PropertyRow
          title={en.AirQualityIndex}
          message={breezometerAqi}
        />
        <PropertyRow
          title={en.DominantPolutant}
          message={dominantPollutantCanonicalName}
        />
        <PropertyRow
          title={en.PolutantDescription}
          message={dominantPollutantDescription}
        />
      </div>
    );
  }

  render() {
    const { color } = { ...this.props.weatherState };

    const title = (
      <h1>
        {this.props.city}
        <span
          className='circle-style'
          style={{
            color,
          }}
        >
          &#8226;
        </span>
      </h1>
    );

    return (
      <Panel header={title} bsStyle="info">
        {this.renderComponentBody()}
      </Panel>
    );
  }
}

export default WeatherQuality;

WeatherQuality.propTypes = {
  weatherState: PropTypes.shape().isRequired,
  city: PropTypes.string.isRequired,
  isFetching: PropTypes.bool.isRequired,
  error: PropTypes.shape(),
};

WeatherQuality.defaultProps = {
  error: null,
};
