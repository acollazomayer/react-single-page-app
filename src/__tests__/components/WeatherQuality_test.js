import React from 'react';
import renderer from 'react-test-renderer';
import WeatherQuality from '../../components/WeatherQuality';

describe('WeatherQuality Component', () => {

  describe('when there is an error', () => {
    const props = {
      isFetching: false,
      city: 'Paris',
      weatherState: {
        color: 'white',
        breezometerAqi: '19',
        breezometerDescription: 'Poor Air Quality',
        dominantPollutantCanonicalName: 'co2',
        dominantPollutantDescription: 'This is risky',
      },
      error: {
        message: 'someerror',
      },
    };
    const wrapper = renderer.create(<WeatherQuality {...props} />);

    it('should render the error message', () => {
      expect(wrapper.toJSON()).toMatchSnapshot();
    });
  });

  describe('When is fetching', () => {
    const props = {
      isFetching: true,
      city: 'Paris',
      weatherState: {
        color: 'white',
        breezometerAqi: '19',
        breezometerDescription: 'Poor Air Quality',
        dominantPollutantCanonicalName: 'co2',
        dominantPollutantDescription: 'This is risky',
      },
    };
    const wrapper = renderer.create(<WeatherQuality {...props} />);

    it('should render the spinner', () => {
      expect(wrapper.toJSON()).toMatchSnapshot();
    });
  });

  describe('When has fetched', () => {
    const props = {
      isFetching: false,
      city: 'Paris',
      weatherState: {
        color: 'white',
        breezometerAqi: '19',
        breezometerDescription: 'Poor Air Quality',
        dominantPollutantCanonicalName: 'co2',
        dominantPollutantDescription: 'This is risky',
      },
    };
    const wrapper = renderer.create(<WeatherQuality {...props} />);

    it('should render the weather quality', () => {
      expect(wrapper.toJSON()).toMatchSnapshot();
    });
  });
});
