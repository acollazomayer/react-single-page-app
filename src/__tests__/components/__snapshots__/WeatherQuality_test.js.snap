// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeatherQuality Component When has fetched should render the weather quality 1`] = `
<div
  className="panel panel-info"
  id={undefined}
>
  <div
    className="panel-heading"
  >
    <h1
      className="panel-title"
    >
      Paris
      <span
        className="circle-style"
        style={
          Object {
            "color": "white",
          }
        }
      >
        •
      </span>
    </h1>
  </div>
  <div
    className="panel-body"
  >
    <div>
      <div
        className="well well-sm"
      >
        <div
          className="col-xs-3"
        >
          <label
            className="control-label"
            htmlFor={undefined}
          >
            Air Quality Message
          </label>
        </div>
        Poor Air Quality
      </div>
      <div
        className="well well-sm"
      >
        <div
          className="col-xs-3"
        >
          <label
            className="control-label"
            htmlFor={undefined}
          >
            Air Quality Index
          </label>
        </div>
        19
      </div>
      <div
        className="well well-sm"
      >
        <div
          className="col-xs-3"
        >
          <label
            className="control-label"
            htmlFor={undefined}
          >
            Dominant Polutant
          </label>
        </div>
        co2
      </div>
      <div
        className="well well-sm"
      >
        <div
          className="col-xs-3"
        >
          <label
            className="control-label"
            htmlFor={undefined}
          >
            Polutant Description
          </label>
        </div>
        This is risky
      </div>
    </div>
  </div>
</div>
`;

exports[`WeatherQuality Component When is fetching should render the spinner 1`] = `
<div
  className="panel panel-info"
  id={undefined}
>
  <div
    className="panel-heading"
  >
    <h1
      className="panel-title"
    >
      Paris
      <span
        className="circle-style"
        style={
          Object {
            "color": "white",
          }
        }
      >
        •
      </span>
    </h1>
  </div>
  <div
    className="panel-body"
  >
    <img
      alt="loading"
      className="spinner"
      src="spinner.svg"
    />
  </div>
</div>
`;

exports[`WeatherQuality Component when there is an error should render the error message 1`] = `
<div
  className="panel panel-info"
  id={undefined}
>
  <div
    className="panel-heading"
  >
    <h1
      className="panel-title"
    >
      Paris
      <span
        className="circle-style"
        style={
          Object {
            "color": "white",
          }
        }
      >
        •
      </span>
    </h1>
  </div>
  <div
    className="panel-body"
  >
    someerror
  </div>
</div>
`;
